airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -425,6 +439,10 @@ public List<AutoCloseableIterator<AirbyteMessage>> getIncrementalIterators(final
           AutoCloseableIterators.concatWithEagerClose(AirbyteTraceMessageUtility::emitStreamStatusTrace, snapshotIterator,
               AutoCloseableIterators.lazyIterator(incrementalIteratorSupplier, null)));
 
+    } else if (PostgresUtils.isXmin(sourceConfig) && isIncrementalSync(catalog)) {
+      final XminStateManager xminStateManager = new XminStateManager(stateManager.getRawStateMessages());
+      final PostgresXminHandler handler = new PostgresXminHandler(database, sourceOperations, getQuoteString(), xminStatus, xminStateManager);
```

Comment:
There is a chance that xminStatus could be null at this point right?

Reply:
Actually, we are saving `xminStatus` in `logPresyncDebugData`, which runs prior to `getIncrementalIterators()` so this will not be null